### PR TITLE
Performance improvement in ProtobufTranslator

### PR DESCRIPTION
**Pull request by oliviertoupin**
- Using prepend and a reverse list for encoding
- Same for decoding but use it's own for clarity

### CODE REVIEW

File: mry-core/src/main/scala/com/wajam/mry/api/protobuf/ProtobufTranslator.scala [hunk 1]
```diff
@@ -114,7 +115,7 @@ private class InternalProtobufTranslator {
 
     val instanceHeapId = currentHeapId
 
-    tempHeap = tempHeap :+ mryData
+    reverseEncodingHeap = mryData +: reverseEncodingHeap
```
Comment (alexbergeron) [inline mry-core/src/main/scala/com/wajam/mry/api/protobuf/ProtobufTranslator.scala]: Use `::`, it's the official list prepend method (and kinda faster from what I've seen in Scala code)

File: mry-core/src/main/scala/com/wajam/mry/api/protobuf/ProtobufTranslator.scala [hunk 2]
```diff
@@ -148,15 +151,17 @@ private class InternalProtobufTranslator {
           case pMryData if pMryData.hasValue => pMryData.getValue
         }
 
-        tempHeap = tempHeap :+ value
+        reverseDecodingHeap = value +: reverseDecodingHeap
```
Comment (alexbergeron) [inline mry-core/src/main/scala/com/wajam/mry/api/protobuf/ProtobufTranslator.scala]: `::`

**Comment by coderunner** [comment]
Looks good to me +1

**Comment by alexbergeron** [comment]
I still insist on using `::`

**Comment by coderunner** [comment]
+::

**Pull request merged**